### DISCUSSION TITLE
Calculate time spent based on summary

### DISCUSSION THREAD

**Discussion by mrmkrs** [jotaen/klog#305] [Help / Question] [answered]
Hi!
I'm looking for way to show how much time i've spent on items without tags.

Given the following log:

```
2024-04-02
    8:30 - 9:00 #task-a 30min on a
    9:00 - 9:30 #task-b 30 min on b
    9:30 - 10:00 #task-a
    10:00 - 10:15 only a summary here
    11:00 - 11:30 only a summary here
```

Most ideal would be that I could get a report that said 'only a summary here    30m' (so the two entries summed up). I can get that information for tags using `klog tags` but I couldn't find a way to report anything that only has a summary. Is something like that supported?

**Comment by jotaen** [comment]
I don’t believe this is possible right now. May I ask what your specific use-case is? Maybe there might be a workaround.

**Comment by mrmkrs** [comment]
Hi,

So what I'm basically looking for is a sum of time for everything that has no tag. I couldn't find any command that would tell me how much time was spent a certain log. Aggregation of the same summary would be a nice to have :)

I've always used https://gtimelog.org/ which allowed me to log what i've done, instead 'log what you are going to do'. I find this workflow far more intuitive and better matches my workflow. Throughout the day I can log 'did x'... 'did y', .. 'did x'. In the report I could see 'time spent on non-categorized tasks: x hours'. I really hope I can get that info somehow from klog. The workaround would be to obviously always use tags.

**Comment by jotaen** [comment, marked as answer]
I think what might be doable would be a filter flag like `--no-tag` or `--tag-count 0` or so, but that would then be more for double-checking whether you have forgotten to add a tag somewhere. It wouldn’t distinguish the resulting entries any further (e.g., based on the summary text).

Based on how you describe your use-case, I think that’s actually what tags were meant to solve in klog. Note that tags are designed to naturally fit into the flow of text (much like e.g. hashtags on Twitter):

```
2000-01-01
  3h20m Worked on a #task
  15:00-16:30 Had #lunch with Mike
  1h15m Went to the #gym
```

The idea is for tags to get as little into your way as possible when writing the summary text, while still allowing you to categorise your data in a rigorous and logical way. Aggregating summary text would have some shortcomings here, because e.g. you have to watch out for spelling differences:

```
2000-01-01
No tags → would aggregate into 3 separate items
  1h Worked on a task
  1h worked on task
  1h task
```

```
2000-01-01
With tags → aggregates into 1 combined item
  1h Worked on a #task
  1h worked on #task
  1h #task
```

**Comment by mrmkrs** [comment]
Hi thank you for your explanation! Working with tags will do just fine I think. But than there is still the issue that any task without tag is never reported. If you log items with the exact time spent, than it's easy to see what you've done. But for time ranges it's more difficult:

2000-01-01
  10:02 - 14:37 Worked on a task
  15:00-16:30 Had #lunch with Mike
  18:00-18:30 Had a late #lunch
  30m Writing stuff

The only way to know how much time I've spent on that task is to calculate it myself. Maybe some option in `klog print` to also print the time spent for time ranges per line? It doesn't matter if it has tags or not:

2000-01-01
  10:02 - 14:37 Worked on a task (4h35m)
  15:00-16:30 Had #lunch with Mike (1h30m)
  18:00-18:30 Had a late #lunch (30m)
  30m Writing stuff

**Reply by jotaen** [reply]
You could try `klog print --with-totals` for that.

Note that there is a tiny bug right now, though, because `--with-totals` colors the left-hand totals in red, but they should actually be grey. This is already fixed on the main branch, but not released yet.

**Reply by mrmkrs** [reply]
Nice, I totally missed that argument! Thank you for your help.
